tide_landing_page.module

```diff
@@ -111,18 +111,18 @@ function tide_landing_page_form_node_form_alter(&$form, FormStateInterface $form
         }
       }
     }
-    // Add conditional field for show table of content.
-    if (isset($form['field_node_display_headings'])) {
-      $form['field_node_display_headings']['#states'] = [
-        'visible' => [
-          ':input[name="field_show_table_of_content[value]"]' => [
-            'checked' => TRUE,
-          ],
+  }
+  // Add conditional field for show table of content.
+  if (isset($form['field_node_display_headings']) && isset($form['field_show_table_of_content'])) {
+    $form['field_node_display_headings']['#states'] = [
+      'visible' => [
+        ':input[name="field_show_table_of_content[value]"]' => [
+          'checked' => TRUE,
         ],
-      ];
-      if (!isset($form['field_node_display_headings']['widget']['#default_value'])) {
-        $form['field_node_display_headings']['widget']['#default_value'] = 'showH2';
-      }
+      ],
+    ];
+    if (!isset($form['field_node_display_headings']['widget']['#default_value'])) {
```

Comment:
Do you want isset here or empty? Is there a possibility that #default_value key is always set but the value is empty?

Reply:
That is a good question @asarfara , 
the whole logic from line 124 to line 125 could be deleted. cc @MdNadimHossain

Reply:
@vincent-gao there was an issue on existing content, where "show table of content" was ticked but the default value wasn't set. This check is for that. Also, Ali is right, please make it an empty check, rather than checking if the field is set.

Reply:
great! thanks